src/components/TokenSelector.tsx

```diff
@@ -331,17 +366,33 @@ const TokenSelector: React.FC<Props> = ({ isDisabled, tokens, selected, onChange
   // When the search input is focused, force menu to remain open
   const onMenuInputFocus = useCallback(() => setIsFocused(true), [])
 
-  const onKeyDown = useCallback((e): void => {
-    if (e.key === 'Escape') {
-      // Close menu on `Escape`
-      e.stopPropagation()
-      setIsFocused(false)
-    } else if (!e.target.value && e.key === ' ') {
-      // Prevent a space when input in empty.
-      // That closes the menu. (/shrug)
-      e.preventDefault()
-    }
-  }, [])
+  const onKeyDown = useCallback(
+    (e: React.KeyboardEvent<HTMLInputElement> & { target: HTMLInputElement }): void => {
+      if (e.key === 'Escape') {
+        // Close menu on `Escape`
+        e.stopPropagation()
+        setIsFocused(false)
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      } else if (!e.target.value! && e.key === ' ') {
+        // Prevent a space when input in empty.
+        // That closes the menu. (/shrug)
+        e.preventDefault()
+      } else if (e.key === 'Enter' && networkId && isAddress(e.target.value.toLowerCase())) {
+        const tokenAddress = e.target.value
+        if (tokenListApi.hasToken({ tokenAddress, networkId })) return
+
+        // prevents double-catch of this event
+        // double because it's captured from onKeyDown in MenuList
+        // and in general on Select, I guess
+        e.stopPropagation()
```

Comment:
oof all these stopPropagations etc are getting messy. This is a `react-select` problem I'm assuming?

Reply:
yes, it handles event like a blackbox, so these issues pop up from time to time